### DISCUSSION TITLE
Can't use union with a type and a list

### DISCUSSION THREAD

**Discussion by JoseDRojasA** [graphql-java/graphql-java#2961] [General]
**Describe the bug**
I used to have the following schema:
`
type Cat {
  name: String
}

 type Query {
  getCats: [Cat]
}
`

But We want to send also the total amount of cats but use the union types feature to avoid a breaking change in the API.

So the new schema would be:

`
type Cat {
  name: String
}

type CatWithDetails {
  cats:  [Cat]
  totalCats: Int
}

union CatUnion = CatWithDetails |  Cat[]

 type Query {
  getCats: CatUnion
}
`

And when I start the project, I get this error message:

`
java.lang.IllegalStateException: Failed to load ApplicationContext
Caused by: org.springframework.beans.factory.BeanCreationException: 
Error creating bean with name 'graphQLProvider': Invocation of init method failed; nested exception is SchemaProblem{errors=[InvalidSyntaxError{ message=Invalid Syntax : offending token '[' at line 23 column 80 ,offendingToken=[ ,locations=[SourceLocation{line=23, column=80}] 
}

union CatUnion = CatWithDetails |  Cat[]

type Query {
  getCats: CatUnion
}]}

`

Am I missing something?

Thank you in advance

**Comment by andimarek** [comment]
a union can only consists of object types, not list of them.
so 
```graphql
union CatUnion = CatWithDetails | Cat[]
```
is actually an invalid syntax because `[]` is not allowed after `Cat`.
```graphql
union CatUnion = CatWithDetails | Cat
```
would be valid.
